sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -200,8 +200,30 @@ private static string DetermineValidPathCharacters()
         /// <returns></returns>
         public static string GetExtension(string path)
         {
-            if (path == null)
+            if (path is null)
                 return null;
+
+#if NET8_0_OR_GREATER
+            // LastIndexOf and LastIndexOfAny is vectorized on .NET8+ and is
+            // signifigantly faster for cases where 'path' does not end with a short file
+            // extension, such as GUIDs
+            int extensionIndex = path.AsSpan().LastIndexOf('.');
```

Comment:
Would it be cleaner to assign the span to a local and then use it through the method?